src/js/libgeo/Prover.js

```diff
@@ -68,21 +74,77 @@ function incidentPC(p, c) {
             var erg = General.mult(c.matrix, p.homog);
             erg = General.mult(p.homog, erg);
             erg = CSNumber.abs(erg);
-            if (erg.value.real < 0.0000000000001) {
-                p.incidences.push(c.name);
-                c.incidences.push(p.name);
-            }
+            return (erg.value.real < 0.0000000000001);
         }
     };
 }
 
 function checkConjectures() {
+    var debug = false;
+    if (debug) console.log("conjectures", conjectures.length);
+    //if (!debug)
     if (conjectures.length === 0) return;
-    // TODO: we need some randomized proving here:
-    // move free elements and check conjectures a number of times.
-    // For now assume that all conjectures could be verified.
+    backupGeo();
+
+    var ii, jj, emove;
+    var nummoves = 3;
+
+    // filter free objects which are involved in conjectures
+    var involved;
+
+    var recalcInvolved = function() {
+        involved = [];
+        conjectures.forEach(function(con) {
+            var invs = con.getInvolved();
+            invs = invs.forEach(function(el) {
+                if (involved.indexOf(el) < 0 && el.movable) involved.push(el);
```

Comment:
I think `involved` should also cover non-movable elements, and should be computed transitively. It may well be that some conjecture directly involves two non-free elements, but in that case we need to consider all the free and semi-free ancestors up the `args` connections, to find which things we should move to affect the given conjecture.